crates/bevy_ecs/src/system/commands/parallel_scope.rs

```diff
@@ -0,0 +1,94 @@
+use std::cell::Cell;
+
+use thread_local::ThreadLocal;
+
+use crate::{
+    entity::Entities,
+    prelude::World,
+    system::{SystemParam, SystemParamFetch, SystemParamState},
+};
+
+use super::{CommandQueue, Commands};
+
+#[derive(Default)]
+/// The internal [`SystemParamState`] of the [`ParallelCommands`] type
+pub struct ParallelCommandsState {
+    tls: ThreadLocal<Cell<CommandQueue>>,
+}
+
+/// An alternative to [`Commands`] that can be used in parallel contexts, such as those in [`Query::par_for_each`]
+///
```

Comment:
One more thought: probably should leave a note here saying that command application order is dependent on the number of threads, so any non-idempotent commands may result in different results on different machines.

Reply:
Did you mean "non-commutative"? Given that commands take `self` and can't be executed twice idempotence is a bit of a moot point.

Reply:
Sorry, was thinking of this through the lens of networking where command application like this is a potential concern for determinism. Yes you're right.